test/NuGet.Core.Tests/NuGet.Build.Tasks.Test/StaticGraphRestoreArgumentsTests.cs

```diff
@@ -0,0 +1,348 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using FluentAssertions;
+using Xunit;
+
+namespace NuGet.Build.Tasks.Test
+{
+    public class StaticGraphRestoreArgumentsTests
+    {
+        /// <summary>
+        /// Verifies that the <see cref="StaticGraphRestoreArguments.Write(TextWriter)" /> and <see cref="StaticGraphRestoreArguments.Read(TextReader)" /> support global properties with complex characters like line breaks, XML, and JSON.
+        /// </summary>
+        [Fact]
+        public void Read_WhenGLobalPropertiesContainComplexCharacters_CanBeRead()
+        {
+            var expected = new StaticGraphRestoreArguments
+            {
+                GlobalProperties = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
+                {
+                    ["PublishDir"] = @"D:\RPublish",
+                    ["SolutionFileName"] = "IPIPE.sln",
+                    ["LangName"] = "en-US",
+                    ["PublishTrimmed"] = "false",
+                    ["CurrentSolutionConfigurationContents"] = @"<SolutionConfiguration>
+<ProjectConfiguration Project=""{10192869-f315-4bd4-af9b-2a20cb8a6afa}"" AbsolutePath=""E:\TFS\IPIPE\Libraries\Recon\ipipe.Recon.BackProjectors\ipipe.Recon.BackProjectors.csproj"">Release|x64</ProjectConfiguration>
+<ProjectConfiguration Project=""{e854317c-0c24-4927-beef-e2c3b2c2fc9f}"" AbsolutePath=""E:\TFS\IPIPE\Libraries\Managed\siemens.CardioTools\ipipe.CardioTools.csproj"">Release|x64</ProjectConfiguration>
```

Comment:
Should we replace `siemens` with something like `contoso` just in case? It's a proprietary name.